### PR TITLE
Added tests and lowercased enums

### PR DESCRIPTION
**Pull request by masters3d**
* Added test and lowercased enums
https://bugs.swift.org/browse/SR-1187

### CODE REVIEW

File: Sources/PlayingCard.swift [hunk 1]
```diff
@@ -31,13 +31,13 @@ public func ==(lhs: PlayingCard, rhs: PlayingCard) -> Bool {
 extension PlayingCard: Comparable {}
 
 public func <(lhs: PlayingCard, rhs: PlayingCard) -> Bool {
-    return lhs.suit < rhs.suit || (lhs.suit == rhs.suit && lhs.rank < rhs.rank)
+    return lhs.rank == rhs.rank ? lhs.suit == rhs.suit : lhs.rank < rhs.rank
```
Comment (ddunbar) [inline Sources/PlayingCard.swift]: This isn't equivalent, this is no longer properly comparing cards when the rank is the same but the suit is different.

File: Sources/PlayingCard.swift [hunk 2]
```diff
@@ -31,13 +31,13 @@ public func ==(lhs: PlayingCard, rhs: PlayingCard) -> Bool {
...
 }
 
 // MARK: - CustomStringConvertible
 
 extension PlayingCard : CustomStringConvertible {
     public var description: String {
-        return "\(suit)\(rank)"
+        return "\(rank)\(suit)"
```
Comment (ddunbar) [inline Sources/PlayingCard.swift]: Why did you switch the order here?

File: Tests/PlayingCard/CardTests.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+*/
+
+@testable import PlayingCard
+import XCTest
+
+class CardTests: XCTestCase {
+    
+    func testCardSingle(){
```
Comment (ddunbar) [inline Tests/PlayingCard/CardTests.swift]: Space after `()` please, as in:
```
    func testCardSingle() {
 ```

File: Tests/PlayingCard/SuitTests.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@testable import PlayingCard
+import XCTest
+
+class SuitTests: XCTestCase {
+    
+    func testSuitStringEquality(){
+        let c:Suit = .clubs
+        let d:Suit = .diamonds
+        let h:Suit = .hearts
+        let s:Suit = .spades
```
Comment (ddunbar) [inline Tests/PlayingCard/SuitTests.swift]: I would write these inline, there isn't a need to put them in separate variables.

File: Tests/PlayingCard/XCTestManifests.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension CardTests{
```
Comment (ddunbar) [inline Tests/PlayingCard/XCTestManifests.swift]: Space after extension name please, as in:
```
extension CardTests {
}
```

**Comment by ddunbar** [comment]
@mattt Do you have an opinion on extending these repos by adding tests? I would rather not invest very much time in the maintenance of them, but tests are also a best practice.

File: Sources/PlayingCard.swift (hunk 1 above)
Comment (masters3d) [inline Sources/PlayingCard.swift]: ah yes. I fixed this and added a new test to confirm.

File: Sources/PlayingCard.swift (hunk 2 above)
Comment (masters3d) [inline Sources/PlayingCard.swift]: I didn't make sense to me to have the suit first but I noticed that it prints better that way so I reverted it back.

**Comment by mattt** [comment]
@ddunbar Because this is code used in a tutorial, the only thing that really matters is that the example works as intended. I think it'd only really make sense to add tests if 1) automated testing were something covered by the tutorial and 2) all packages in the example were tested. If you're interested in expanding the scope of the tutorial, I'd be happy to coordinate with you.

File: Sources/PlayingCard.swift [hunk 6]
```diff
@@ -31,13 +31,13 @@ public func ==(lhs: PlayingCard, rhs: PlayingCard) -> Bool {
 extension PlayingCard: Comparable {}
 
 public func <(lhs: PlayingCard, rhs: PlayingCard) -> Bool {
-    return lhs.rank == rhs.rank ? lhs.suit == rhs.suit : lhs.rank < rhs.rank
+    return lhs.rank == rhs.rank ? lhs.suit < rhs.suit : lhs.rank < rhs.rank
```
Comment (mattt) [inline Sources/PlayingCard.swift]: I would prefer if this were reverted this to the original logic. This new behavior, while concise, does not reflect how most people would reason about comparing playing cards.

**Comment by ddunbar** [comment]
There are patches for tests for all packages, and expanding the tutorial to cover this would probably make sense. I'll go ahead and plan to iterate and merge these and we can figure out how/when to update the tutorial.

File: Sources/PlayingCard.swift (hunk 6 above)
Comment (masters3d) [inline Sources/PlayingCard.swift]: the original logic allows for A<K in some cases.
image

**Comment by mxcl** [comment]
The lesson here is: keep PRs focussed, since we can't merge the lowercase enums until the stuff Daniel wants is done.

**Comment by masters3d** [comment]
@mxcl noted. I am happy to change those back to upper case and/or help update the tutorial.

**Pull request merged**